manager/src/main/java/com/redhat/service/bridge/manager/models/Processor.java

```diff
@@ -28,8 +31,15 @@
 @NamedQueries({
         @NamedQuery(name = "PROCESSOR.findByBridgeIdAndName",
                 query = "from Processor p where p.name=:name and p.bridge.id=:bridgeId"),
-        @NamedQuery(name = "PROCESSOR.findByStatusesAndShardId",
-                query = "from Processor p join fetch p.bridge where p.status in (:statuses) and p.bridge.status='AVAILABLE' and p.shardId=:shardId"),
+        @NamedQuery(name = "PROCESSOR.findByStatusesAndShardIdWithReadyDependencies",
+                query = "select p " +
+                        "from Processor p " +
+                        "join fetch p.bridge " +
+                        "left join p.connectorEntities as c " +
+                        "where p.status in (:statuses) " +
+                        "and p.bridge.status='AVAILABLE' " +
+                        "and p.shardId=:shardId " +
+                        "and (c is null or c.status = 'READY')"),
```

Comment:
Do not we have some enums for AVAILABLE and READY? just asking as saw some PR where some enums were renamed. So maybe using enums here could help if some status is renamed in future?

Reply:
We have a generic `BridgeStatus` enum for the entities having the same lifecycle